sql/core/src/test/scala/org/apache/spark/sql/test/TestSQLContext.scala

```diff
@@ -61,7 +61,10 @@ private[sql] object TestSQLContext {
   val overrideConfs: Map[String, String] =
     Map(
       // Fewer shuffle partitions to speed up testing.
-      SQLConf.SHUFFLE_PARTITIONS.key -> "5")
+      SQLConf.SHUFFLE_PARTITIONS.key -> "5",
+      // Enable parquet read field id for tests to ensure correctness
+      SQLConf.PARQUET_FIELD_ID_READ_ENABLED.key -> "true"
```

Comment:
Does this mean that we will not test match by name and will always test by field id?

Reply:
Not really. Again, enabling this flag would only try to match field ids `if there exists any`, but disabling this flag will completely ignore matching using field id.

I wanted to enable this flag for all tests to detect any regressions in existing test cases, in case that in the future this flag is turned on by default.

Reply:
But they would exist once we start writing field ids for all of the fields, would not they?

Reply:
Oh I see what you meant, let me just grid test this for the parquet suites only.